geoindicators/src/main/groovy/org/orbisgis/SpatialUnits.groovy

```diff
@@ -202,22 +205,63 @@ static IProcess prepareRSUData(){
  */
 static IProcess createBlocks(){
     return processFactory.create("Merge the geometries that touch each other",
-            [inputTableName: String, distance : double, prefixName: String, datasource: JdbcDataSource],
+            [inputTableName: String, distance : 0.0d, prefixName: "block", datasource: JdbcDataSource],
             [outputTableName : String, outputIdBlock: String],
-            { inputTableName,distance =0.0, prefixName="block", datasource ->
-                logger.info("Merging the geometries...")
-
+            { inputTableName,distance, prefixName, JdbcDataSource datasource ->
+                logger.info("Creating the blocks...")
                 def columnIdName = "id_block"
 
                 // The name of the outputTableName is constructed
                 String baseName = "created_blocks"
                 String outputTableName = prefixName + "_" + baseName
 
-                datasource.execute "DROP TABLE IF EXISTS $outputTableName"
-                datasource.execute "CREATE TABLE $outputTableName as  select  EXPLOD_ID as $columnIdName, the_geom "+
-                        "from st_explode ('(select ST_UNION(ST_ACCUM(ST_BUFFER(THE_GEOM,$distance))) as the_geom"+
-                        " from $inputTableName)')"
-                logger.info("The geometries have been merged")
+
+
+                //Find all neighbors for each building
+                logger.info("Building index to perform the process...")
+                if(!datasource.getSpatialTable(inputTableName).the_geom.indexed){
+                    datasource.getSpatialTable(inputTableName).the_geom.createSpatialIndex()
```

Comment:
Indexes are created only if they not already exists so you don't need to test that before.

Reply:
I will fix it